sql/executor.go

```diff
@@ -83,7 +83,15 @@ func newExecutor(db client.DB, gossip *gossip.Gossip, leaseMgr *LeaseManager) *E
 		leaseMgr: leaseMgr,
 	}
 	exec.systemConfigCond = sync.NewCond(&exec.systemConfigMu)
-	gossip.RegisterSystemConfigCallback(exec.updateSystemConfig)
+
+	configC := make(chan *config.SystemConfig, 4)
+	go func() {
```

Comment:
Avoid bare goroutines. If a goroutine doesn't stop when the stopper says it's time to stop, it may find itself unable to complete its last iteration and get caught by leaktest. This should be a `stopper.RunWorker` and select on `stopper.ShouldStop()` (and since it's no longer a `for range` you don't need to close the channel on the gossip side).